src/lib/uri/uri.c

```diff
@@ -74,6 +89,7 @@ static void
 uri_param_copy(struct uri_param *dst, const struct uri_param *src)
 {
 	dst->name = xstrdup(src->name);
+	dst->values_capacity = src->values_capacity;
 	dst->value_count = src->value_count;
 	dst->values = (src->value_count == 0 ? NULL :
 		       xmalloc(src->value_count * sizeof(*dst->values)));
```

Comment:
Allocated size does not correspond to the capacity (see also a PoC patch in the issue).

Reply:
Oh, I wasn't conscious enough about it, fixed it (and `uri_copy` too), thanks!